### ISSUE TITLE
[feature request] suggest to add homebrew_site_packages to user_site_packages when current python is not homebrew python

### ISSUE THREAD

**Issue by AlJohri** [Homebrew/homebrew-core#11443]
Currently, homebrew suggests you to add the `homebrew_site_packages` to `#{user_site_packages}/homebrew.pth` if the [following condition](https://github.com/Homebrew/homebrew-core/blob/7c12c5c70505d53881ece6ed430fb5b20a528d0b/Formula/numpy.rb#L60) applies:

```
if build.with?("python") && !Formula["python"].installed?
```

I think the condition should actually be:

```
if build.with?("python") && (!Formula["python"].installed? || `which python` is not homebrew python)
```

Use Case:

I generally use a non-homebrew python (`pyenv`), but tend to use the homebrew packages for numpy, scipy, and matplotlib. However, because it's often a dependency of some formula, I end up having the homebrew python installed in my system.

When I want to get numpy working, I have to add it to my sitepackages folder. I try to `brew info numpy` to find out where the homebrew sitepackages folder is, but I can't find it there so I have to come back to github to look it up.

**tl;dr**: it would be great if it was easier to find the homebrew site packages folder through `brew info xyz` when working with non-homebrew pythons. I think the suggestion above could be a simple fix.

**Comment by AlJohri** [comment]
Alternatively you can ensure `--with-custom-python` appears on all packages that depend on python and that way homebrew python won't get automatically installed via a dependency. That seems like a bit more work though.

**Comment by MikeMcQuaid** [comment]
> tl;dr: it would be great if it was easier to find the homebrew site packages folder through brew info xyz when working with non-homebrew pythons. I think the suggestion above could be a simple fix.

Yep, this seems like a good idea. Could you try and open a pull request? [This document](http://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request.html) should help and we're happy to walk you through anything else.

Thanks!
